pkg/mgrconfig/load.go

```diff
@@ -308,17 +309,28 @@ func (cfg *Config) completeBinaries() error {
 	}
 	cfg.ExecprogBin = targetBin("syz-execprog", cfg.TargetVMArch)
 	cfg.ExecutorBin = targetBin("syz-executor", cfg.TargetArch)
-	// If the target already provides an executor binary, we don't need to copy it.
+
+	if cfg.ExecprogBinOnTarget != "" {
```

Comment:
Why is this bit needed?
I generally looked that you added code for execprog similar to existing executor logic. But this did not exist for executor.

Reply:
`cfg.SysTarget.ExecutorBin` is already used [to indicate that the binary is shipped with the image on e.g. Fuchsia](https://github.com/sephalon/syzkaller/blob/master/sys/targets/targets.go#L563); I simply applied the same mechanism to `ExecprogBin`.